paper.md

```diff
@@ -2,11 +2,77 @@
+    In this paper, we introduce Popper, a container-native workflow engine that executes each step of a workflow in a separate dedicated container and does not assume the presence of a Kubernetes cluster or any cloud based Kubernetes service.
+    With Popper, researchers can build and validate workflows easily in almost any environment of their choice like a local machine, a Slurm based HPC cluster or a Kubernetes based cloud computing environment. 
+    To exemplify the suitability of this workflow engine, we present three case studies where we take examples from Machine Learning and HPC and turn them into Popper workflows.
 ---
 
 # Introduction {#sec:intro}
 
+<!-- defining the problem of reproducibility in computational science -->
+
+Around 48.6% of scientists and researchers working in various domains related to computational science, upload experimental artifacts like code, figures, datasets, configuration files, etc. on open-access repositories like Zenodo [@_zenodo_], Figshare [@_figshare_] or GitHub [@github]. 
+Unfortunately, only 1.1% of the artifacts available online are fully reproducible and 0.6% of them are partially reproducible [@stagge2019assessing]. 
+According to a study of 2016 by Nature, among a group of 1576 scientists around 70% of them failed to reproduce each other's experiments [@baker2016reproducibility].
+This problem occurs mostly due to the lack of proper documentation, missing artifacts, broken software dependencies, etc. 
+Consequently, this results in other researchers wasting time trying to figure out how to reproduce those experiments from the archived artifacts. 
+Hence, it makes the process inefficient and cumbersome leading to what is known as the Reproducibility crisis or Replication crisis in science [@sep-scientific-reproducibility].
+
+![An end-to-end example of a workflow. On the left, we have the `.yml` file that defines the workflow. On the right, a pictorial representation of it.](./figures/casestudy.png){#fig:casestudy}
+
+<!-- discuss previous work -->
+
+Numerous existing research have tried to address the problem of reproducibility [@goodman2016does] in distinct ways like logging and tracing systemcalls, using workflow engines, using correctly provisioned shared and public testbeds, by recording and replaying changes from a stable initial state, etc [@reproducibility2018acm] and these led to the development of various tools and frameworks [@piccolo2016tools] [@peng2011reproducible].
+Scientific workflow engines have been a predominant solution [@stevens2013automated] [@banati2015minimal] [@qasha2016framework] for handling the reproducibility problem by organizing the steps in a complex scientific workflow as the nodes of a directed acyclic graph (DAG) and executing them in correct order [@albrecht2012makeflow].
+Nextflow [@ditommaso_nextflow_2017], Pegasus [@deelman_pegasus_2004] and Taverna [@oinn_taverna_2004] are some popular examples of scientific workflow engines.
```

Comment:
"are examples of widely used scientific workflow engines"?

Reply:
Yeah,  I would try removing all occurences of `some`.